### PR TITLE
feature: Add session type args

### PR DESCRIPTION
**Pull request by AkifumiSato**
#367 

A type argument added to allow users to extend the session type.

### CODE REVIEW

File: src/utils.ts [hunk 1]
```diff
@@ -1,6 +1,6 @@
 import c from "cookie";
 import { ServerResponse } from "http";
-import { Options, SessionData } from "./types";
+import { Options, Session, SessionData } from './types'
```
Comment (hoangvvo) [inline src/utils.ts:3]: Could you respect the current formatting style?

File: src/session.ts [hunk 2]
```diff
@@ -3,10 +3,12 @@ import { IncomingMessage, ServerResponse } from "http";
 import { nanoid } from "nanoid";
 import MemoryStore from "./memory-store";
 import { isDestroyed, isNew, isTouched } from "./symbol";
-import { Options, Session } from "./types";
+import { Options, Session, SessionRecord } from './types'
```
Comment (hoangvvo) [inline src/session.ts:6]: Could you respect the current formatting style?

File: src/session.ts [hunk 3]
```diff
@@ -3,10 +3,12 @@ import { IncomingMessage, ServerResponse } from "http";
...
 import { commitHeader, hash } from "./utils";
 
-export default function session(options: Options = {}) {
+export default function session<T extends SessionRecord = SessionRecord>(options: Options = {}) {
+  type AppSession = Session<T>;
```
Comment (hoangvvo) [inline src/session.ts:10]: type TypedSession might be a better name imo

**Review by hoangvvo — COMMENTED** [review]
Just a few nits. LGTM otherwise.

File: src/utils.ts (hunk 1 above)
Comment (AkifumiSato) [reply src/utils.ts]: Sorry, I fixed.

File: src/session.ts (hunk 2 above)
Comment (AkifumiSato) [reply src/session.ts]: Sorry, I fixed.

**Comment by AkifumiSato** [comment]
@hoangvvo 
Thanks for the review.
Fixed.

**Review by hoangvvo — APPROVED** [review]
LGTM

**Pull request merged**